core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -374,9 +422,10 @@ public FormValidation doCheckPort(@QueryParameter String value) {
         }
 
         @RequirePOST
+        @Restricted(NoExternalUse.class)
         public FormValidation doValidateProxy(
                 @QueryParameter("testUrl") String testUrl, @QueryParameter("name") String name, @QueryParameter("port") int port,
-                @QueryParameter("userName") String userName, @QueryParameter("password") String password,
+                @QueryParameter("userName") String userName, @QueryParameter("password") Secret password,
```

Comment:
Shouldn't it be

```suggestion
                @QueryParameter("userName") String userName, @QueryParameter("secretPassword") Secret password,
```

based on the jelly file?

Reply:
correct, missed in the last change of the value name